odoo/addons/base/models/ir_ui_view.py

```diff
@@ -108,21 +108,33 @@ def transfer_node_to_modifiers(node, modifiers, context=None):
         else:
             modifiers['invisible'] = [('state', 'not in', states)]
 
+    postprocess_context_modifiers = []
     for attr in ('invisible', 'readonly', 'required'):
         value_str = node.attrib.pop(attr, None)
         if value_str:
-            value = bool(quick_eval(value_str, {'context': context or {}}))
+
             if (attr == 'invisible'
                     and any(parent.tag == 'tree' for parent in node.iterancestors())
                     and not any(parent.tag == 'header' for parent in node.iterancestors())):
                 # Invisible in a tree view has a specific meaning, make it a
                 # new key in the modifiers attribute.
-                modifiers['column_invisible'] = value
-            elif value or (attr not in modifiers or not isinstance(modifiers[attr], list)):
+                attr = 'column_invisible'
+
+            # TODO: for invisible="context.get('...')", delegate to the web client.
+            value = quick_eval(value_str, {'context': context}, evaluate=False)
+            if isinstance(value, str):
+                postprocess_context_modifiers.append(attr)
+            else:
+                value = bool(value)
```

Comment:
Is it really worth modifying and going through `quick_eval` at this point? Just duplicate the 4 special cases (to cache their result) then use the fallback for the rest.

Reply:
Agreed. I did not like this change of `evaluate=False` but wasn't sure what to do as alternative.

Instead of the 4 special cases, I decided to use `odoo.tools.str2bool`.
This allows to not re-implement the wheel, but also mean you can now use, in addition to `1`, `0`, `True`, `False`

`invisible="y"`, `invisible="yes"`, `invisible="on", `invisible="t"``

`invisible="n"`, `invisible="no"`, `invisible="off", `invisible="f"``

Not sure if this is such a great idea to add new possibilities in these attributes, and therefore new mess, or not, but it allows to not have to rely on yet another method converting a string to a bool.

Reply:
Sure. Maybe we could update `str2bool` to make its default optionally callable, to avoid the try/except dance?

Reply:
To me the try..except dance is acceptable

Reply:
Alternatively, this might be a use-case for an odoo.tools.Lazy.